src/Interfaces/ManyRelationInterface.php

```diff
@@ -8,10 +8,20 @@
 use Swis\JsonApi\Client\Links;
 use Swis\JsonApi\Client\Meta;
 
+/**
+ * @template TItem of \Swis\JsonApi\Client\Interfaces\ItemInterface
+ */
 interface ManyRelationInterface
 {
+    /**
+     * @param  \Swis\JsonApi\Client\Collection<array-key, TItem>|null  $data
+     * @return static
+     */
     public function setData(?Collection $data);
```

Comment:
Return static also points at a little weirdness, why isn't that type hinted? And should it be?